File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.java
class Solution {
    public int minimumPushes(String word) {
        int arr[]=new int[26];
        for (char ch :word.toCharArray()){
            arr[ch-'a']++;
        }
        Arrays.sort(arr);
        int minkeypress=0;
        int count =0;
        for(int i=25;i>=0;i--){
            if (arr[i]==0) break;
            int val =arr[i]*(count/8 +1);
            minkeypress+=val;
            count ++;
        }
        return minkeypress;
    }

}

File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.py
class Solution(object):
    def minimumPushes(self, word):
        from collections import Counter
        freq=Counter(word)
        sorted_letters = sorted(freq.items(), key=lambda x: -x[1])

        key_press=0
        key_position =0
        press_per_key=[1]*8

        for letter,count in sorted_letters:
            key_index=key_position//8
            pres_count=(key_index%8)+1
            key_press += count * pres_count
            key_position += 1
      
        return key_press




